src/track/beatfactory.cpp

```diff
@@ -1,36 +1,34 @@
-#include <QtDebug>
+#include "track/beatfactory.h"
+
 #include <QStringList>
+#include <QtDebug>
 
-#include "track/beatgrid.h"
-#include "track/beatmap.h"
-#include "track/beatfactory.h"
+#include "track/beats.h"
 #include "track/beatutils.h"
+#include "track/track.h"
 
-BeatsPointer BeatFactory::loadBeatsFromByteArray(const Track& track,
-                                                 QString beatsVersion,
-                                                 QString beatsSubVersion,
-                                                 const QByteArray& beatsSerialized) {
+mixxx::BeatsPointer BeatFactory::loadBeatsFromByteArray(const TrackPointer track,
```

Comment:
Either pass by value or by const ref, but not by const value.

Reply:
Done